server/src/grpc/remote_engine_service/mod.rs

```diff
@@ -155,6 +158,45 @@ impl<Q: QueryExecutor + 'static> RemoteEngineServiceImpl<Q> {
         Ok(Response::new(resp))
     }
 
+    async fn write_batch_internal(
+        &self,
+        request: Request<WriteBatchRequest>,
+    ) -> std::result::Result<Response<WriteResponse>, Status> {
+        let begin_instant = Instant::now();
+        let request = request.into_inner();
+        let mut write_table_futures = Vec::with_capacity(request.batch.len());
+        for one_request in request.batch {
+            let ctx = self.handler_ctx();
+            write_table_futures.push(async move { handle_write(ctx, one_request).await });
+        }
+
+        let handle = self
+            .runtimes
+            .write_runtime
+            .spawn(try_join_all(write_table_futures));
```

Comment:
Shall we spawn L170? Otherwise, the writes for different tables may be sequential?

Reply:
I don't see the advantage about it?